ui/packages/components/src/DatePicker/RangePicker.tsx

```diff
@@ -125,6 +125,11 @@ export const RangePicker = ({
       setStartValid(false);
       setEndValid(false);
     }
+
+    if (upgradeCutoff && absoluteRange?.start && isBefore(absoluteRange.start, upgradeCutoff)) {
+      setStartError('Upgrade required for requested start date');
```

Comment:
**suggestion (copy, non-blocking)**: We don't yet have consistent language for upgrade nudges yet, but I was thinking to tie this into the "history" work that we use on our pricing pages. What do you think?
```suggestion
      setStartError('Please upgrade for increased history limits');
```

Reply:
👍